src/main/java/Duke.java

```diff
@@ -1,10 +1,272 @@
+        separator();
+        System.out.println("Noted. I've removed this task: ");
+        System.out.println(" " + task.getTitle());
+        System.out.println("Now you have " + items.size() + " tasks in the list.");
+        separator();
+    }
+
+    public static void save() {
+        try {
+            // empty current saved items;
+            FileWriter clear = new FileWriter(f);
+            clear.write("");
+            clear.close();
+            int i = 0;
+            for(i = 0; i < Task.numberOfTasks; i++) {
+                Task task = items.get(i);
+
+                FileWriter fw = new FileWriter(f, true);
+
+                fw.write(task.saveFormat() + "\n");
+                fw.close();
+
+            }
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+
+    }
+
+    public static void initData() {
+        try {
+            Scanner s = new Scanner(f); // create a Scanner using the File as the source
+            while (s.hasNext()) {
+                String currentLine = s.nextLine();
+                String[] items = currentLine.split(" | ");
+                String typeOfTask = items[0];
+                boolean isCompleted = items[2].equals("1") ? true : false;
+                // remove "T |"
+                currentLine = currentLine.replace(typeOfTask + " | ", "");
+                // remove completed status
+                currentLine = currentLine.replace(items[2] + " | ", "");
+                if (typeOfTask.trim().equalsIgnoreCase("T")) {
+                    addTodo(currentLine, isCompleted);
+                }
+                if (typeOfTask.trim().equalsIgnoreCase("E")) {
+                    int atIndex = currentLine.indexOf("|");
+                    String title = currentLine.substring(0, atIndex - 1).trim();
+                    String at = currentLine.substring(atIndex + 1).trim();
+
+                    addEvent(title, at, isCompleted);
+                }
+                if (typeOfTask.trim().equalsIgnoreCase("D")) {
+                    int byIndex = currentLine.indexOf("|");
+                    String title = currentLine.substring(0, byIndex - 1).trim();
+                    String by = currentLine.substring(byIndex + 1).trim();
+                    addDeadline(title, by, isCompleted);
+                }
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    public static void markAsDone(int taskId) {
+        Task selectedTask = items.get(taskId - 1);
+        selectedTask.markAsCompleted();
+        separator();
+        System.out.println("Nice! I've marked this task as done: ");
+        System.out.println(" " + selectedTask.getTitle());
+        separator();
+    }
+
+    public static void addTodo(String item, boolean isCompleted) {
+        Task newTask = addTodo(item);
+        if (isCompleted) {
+            newTask.markAsCompleted();
+        }
```

Comment:
Your method name is `addTodo` but you are doing something related to `markAsCompleted` here. This affects the readability and understandability of the code.
You may want to do this as a separate method.